From 1c379cae0198df499ddd2f27b3c28d12af956d4c Mon Sep 17 00:00:00 2001
From: Min Fanlei <minfanlei@wxiat.com>
Date: Tue, 23 Aug 2022 09:18:11 +0800
Subject: [PATCH] sw64: kvm: fix wrong info print of KVM_MEMHOTPLUG
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 1c379cae0198df499ddd2f27b3c28d12af956d4c
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56WV8

--------------------------------

Due to the registration of guest IO address, there is wrong
pr_info of "KVM MEMHOTPLUG support" when booting the guest,
so we fix it.

Signed-off-by: Min Fanlei <minfanlei@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kvm/kvm-sw64.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/sw_64/kvm/kvm-sw64.c b/arch/sw_64/kvm/kvm-sw64.c
index 9d209141820c..06e969caaaa6 100644
--- a/arch/sw_64/kvm/kvm-sw64.c
+++ b/arch/sw_64/kvm/kvm-sw64.c
@@ -308,6 +308,12 @@ int kvm_arch_prepare_memory_region(struct kvm *kvm,
 	if (change == KVM_MR_FLAGS_ONLY || change == KVM_MR_DELETE)
 		return 0;
 
+	if (test_bit(IO_MARK_BIT, &(mem->guest_phys_addr)))
+		return 0;
+
+	if (test_bit(IO_MARK_BIT + 1, &(mem->guest_phys_addr)))
+		return 0;
+
 #ifndef CONFIG_KVM_MEMHOTPLUG
 	if (mem->guest_phys_addr) {
 		pr_info("%s, No KVM MEMHOTPLUG support!\n", __func__);
@@ -315,12 +321,6 @@ int kvm_arch_prepare_memory_region(struct kvm *kvm,
 	}
 #endif
 
-	if (test_bit(IO_MARK_BIT, &(mem->guest_phys_addr)))
-		return 0;
-
-	if (test_bit(IO_MARK_BIT + 1, &(mem->guest_phys_addr)))
-		return 0;
-
 	if (!sw64_kvm_pool)
 		return -ENOMEM;
 
-- 
2.33.0

